### PR TITLE
Issue #3232719 by tBKoT: Revert duplicate disable event enrolments functionality

### PR DESCRIPTION
**Pull request by dmytriikaiun**
## Problem
In [PR #2283](https://github.com/goalgorilla/open_social/pull/2283) we introduced a new field `field_event_enable_enrollment` but I can’t from the PR understand what the difference is with `field_event_enroll`?

One is “Enable event enrollment” and the other is “Allow users to enroll in the event” So it appears we duplicated functionality here?

There’s also `field_enroll_method` to which we could also add a disabled state then we wouldn’t even need `field_event_enroll` anymore either.

## Solution
- Revert PR 2283 entirely
- Include an update hook to remove the new field field_event_enable_enrollment.
- Then fix the original bug with the fields that already existed (field_event_enroll or field_enroll_method). 

## Issue tracker
https://www.drupal.org/project/social/issues/3232719
https://getopensocial.atlassian.net/browse/YANG-6032

## How to test
- [ ] Go to the event settings page `admin/config/opensocial/event` as a sitemanager
- [ ] You should be able to enable/disable option `Allow event organizers to disable event enrolments`
- [ ] If disable all event enrollment buttons should not display
- [ ] If enabled event organizers can decide to enable/disable event enrollment

## Screenshots
N/A

## Release notes
SM can disable event enrollments platform-wide and event organizers could disable event enrollment if allowed to.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2489-xyr74mnet6vubt3u4fkbiizhwjmgg3jk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6138ba5b08cf4761a96001bb

File: modules/social_features/social_event/social_event.install [hunk 1]
```diff
@@ -1800,66 +1800,83 @@ function social_event_update_8915() {
 }
 
 /**
- * Add the 'enable event enrolment' field.
- */
-function social_event_update_8916() {
```
Comment (ribel) [inline modules/social_features/social_event/social_event.install:1805]: We can't just remove update hooks that were released before.
You can remove the body of the update hook if it is deprecated, but leave a clear comment on what was done. 
See an example in `social_core_update_10102()`

File: modules/social_features/social_event/social_event.install [hunk 2]
```diff
@@ -1800,66 +1800,83 @@ function social_event_update_8915() {
...
-  $updateHelper->executeUpdate('social_event', 'social_event_update_8917');
+  $updateHelper->executeUpdate('social_event', 'social_event_update_8919');
 
   // Output logged messages to related channel of update execution.
   return $updateHelper->logger()->output();
 }
 
 /**
- * Set default configs for event.
+ * Update event values.
```
Comment (ribel) [inline modules/social_features/social_event/social_event.install:1817]: We already have released an update hook with a bigger number: `social_event_update_10201`.
It means this update hook will never be run on sites that use OS 10.2.x and later.
I think we need to add this PR to the 10.3 milestone. It means you can use update hooks bigger than `social_event_update_10304`.

**Review by ribel — CHANGES_REQUESTED** [review]
@dmytriikaiun 
Please check the failing tests and fix update hooks.

**Comment by ribel** [comment]
@dmytriikaiun Also please:

1. Create a drupal.org issue
2. Create drupal.org change record
3. Update How to test (label and behaviour is different)

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2489-xyr74mnet6vubt3u4fkbiizhwjmgg3jk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6138ba5b08cf4761a96001bb

**Review by ribel — APPROVED** [review]

**Comment by ribel** [comment]
@dmytriikaiun Thanks for the fixes, but I still see failing test related to the update hook:

```
Error: Call to a member function getValue() on null in social_event_update_10305() (line 1950 of /var/www/html/profiles/contrib/social/modules/social_features/social_event/social_event.install).
```

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2489-xyr74mnet6vubt3u4fkbiizhwjmgg3jk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6138ba5b08cf4761a96001bb

**Comment by ribel** [comment]
@dmytriikaiun We need to fix (or maybe deprecate) `social_event_post_update_10301_enable_event_enrollment()`, because it fails on update scenario.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2489-xyr74mnet6vubt3u4fkbiizhwjmgg3jk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6138ba5b08cf4761a96001bb

**Comment by navneet0693** [comment]
@dmytriikaiun Please resolve the merge conflict.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2489-xyr74mnet6vubt3u4fkbiizhwjmgg3jk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6138ba5b08cf4761a96001bb

**Review by navneet0693 — APPROVED** [review]

**Comment by ribel** [comment]
🍒 picked to 10.3.x

**Pull request merged**